### PR TITLE
Fix new metrics for scenarios

### PR DESCRIPTION
**Pull request by lego963**
Fix: #108

### CODE REVIEW

**Comment by lego963** [comment]
codecov uploading error was fixed by update dep

File: delatore/outputs/__init__.py [hunk 1]
```diff
@@ -11,6 +11,6 @@
 
 async def start_outputs(service: Service, stop_event: Event, config: InstanceConfig):
     await asyncio.wait([
-        AlertaRunner(service, stop_event, config).start(),
+        # AlertaRunner(service, stop_event, config).start(),
```
Comment (anton-sidelnikov) [inline delatore/outputs/__init__.py:14]: Why commented?

Comment (lego963) [reply delatore/outputs/__init__.py:14]: debug mode, forgot about alerta

File: delatore/configuration/dynamic.py [hunk 2]
```diff
@@ -32,7 +32,15 @@ def read_config(config_file=RESOURCES['config.ini']):
         alerta_api_key = defaults.get('alerta_api_key', os.getenv('alerta_api_key'))
         alerta_service = defaults.get('alerta_service', os.getenv('alerta_service'))
         proxy = defaults.get('proxy', None)
-        return InstanceConfig(token, chat_id, influx_password, awx_auth_token, alerta_api_key, alerta_service, proxy)
+        return InstanceConfig(
```
Comment (outcatcher) [inline delatore/configuration/dynamic.py:35]: With so many arguments it makes sense to use argument names (`arg1=vall`)

File: delatore/sources/influx.py [hunk 3]
```diff
@@ -296,21 +297,16 @@ async def _get_status(self, metric):
...
         statuses = {}
         for series in last_record.raw['series']:
             try:
                 hostname = series['tags']['server']
-            except KeyError:
-                continue
-            try:
-                query_time = series['values'][0][0]
-                last_time_ms = _convert_time(query_time)
+                last_time = series['values'][0][0]
```
Comment (outcatcher) [inline delatore/sources/influx.py:308]: `last_time, last_timems, * = series[values][0]` can be put in try... block alone

Comment (lego963) [reply delatore/sources/influx.py:305]: we will obtain less readable code

Comment (outcatcher) [reply delatore/sources/influx.py:305]: then just move `series['values'][0]` to vaiable

File: delatore/outputs/telegram/bot.py [hunk 4]
```diff
@@ -131,7 +128,7 @@ async def start_posting(self):
             message = await self.client.get(.1)
             if message is not None:
                 data = json.loads(message)
-                if (len(data['status_list'])) > 0 and ('status_list' in data):
+                if 'status_list' in data and len(data['status_list']) > 0:
```
Comment (outcatcher) [inline delatore/outputs/telegram/bot.py:131]: `len(data['status_list']) > 0` can be simplified to `data['status_list']`

Comment (lego963) [reply delatore/outputs/telegram/bot.py]: fixed

File: delatore/sources/__init__.py [hunk 5]
```diff
@@ -18,18 +20,18 @@ async def start_sources(msg_service: Service, stop_event: asyncio.Event, config:
     await asyncio.wait([
         AWXApiSource(msg_service.get_client(), config).start(stop_event),
         AWXWebHookSource(msg_service.get_client()).start(stop_event),
-        start_influx_sources(msg_service, stop_event, config )
+        start_influx_sources(msg_service.get_client(), stop_event, config)
     ])
 
 
-async def start_influx_sources(msg_service: Service, stop_event: asyncio.Event, config: InstanceConfig):
+async def start_influx_sources(msg_service: Client, stop_event: asyncio.Event, config: InstanceConfig):
```
[the remark this answers is not in the document]
Comment (lego963) [reply delatore/sources/__init__.py]: outdated file

File: delatore/sources/influx.py [hunk 6]
```diff
@@ -349,108 +338,130 @@ async def _get_status(self, metric):
...
+            host = series['tags']['host']
             values = series['values'][0]
             last_time = values[0]
             self._fail_count.append(values[1])
         except (KeyError, IndexError):
             return generate_status(metric.name, Status.NO_DATA, None)
 
-        if min(self._fail_count) > self.threshold:  # if there are at least one error per minute for 5 minutes
-            return generate_error(metric.name, self._get_error_message(hostname))
+        if min(self._fail_count) > self._threshold_size:  # if there are at least one error per minute for 5 minutes
```
[the remark this answers is not in the document]
Comment (lego963) [reply delatore/sources/influx.py]: it's here `self._threshold_size`

**Pull request merged**